### PR TITLE
Fix Various Issues

### PR DESCRIPTION
**Pull request by RlonRyan**
As to correct things.

### CODE REVIEW

**Comment by RlonRyan** [comment]
@InfinityRaider Please review this and push agricraft-2.0.0-i0.3.0-a4 as a CurseForge alpha if you approve. We have too many downloads on an alpha that turned out to be rather non-functional...

File: src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java [hunk 1]
```diff
@@ -21,6 +21,9 @@
     public BlockCondition(FuzzyStack stack, BlockRange range) {
         this.amount = stack.toStack().stackSize;
         this.volume = range.getVolume();
+        if(this.amount > 1) {
+            throw new IndexOutOfBoundsException("The required amount of blocks must be greater than zero!");
```
Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: Don't like this, can be made sure in the consructor...

Comment (RlonRyan) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: Wait what? That is the constructor...?

Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: I mean the FuzzyStack constructor, make sure the size is never <= 0

Comment (RlonRyan) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: Yeah that's a good idea.

File: src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java [hunk 2]
```diff
@@ -21,6 +21,9 @@
...
+        }
         if(this.amount > this.volume) {
             throw new IndexOutOfBoundsException("Required amount of blocks exceeds volume of range!");
```
Comment (RlonRyan) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: So we catch them early.

Comment (InfinityRaider) [inline src/main/java/com/infinityraider/agricraft/api/requirement/BlockCondition.java]: Ok then...

**Pull request merged**